### PR TITLE
Bug fixes and warning fixes for test_dataloader.c

### PR DESCRIPTION
**Pull request by rosslwheeler**
- test_dataloader.c - In moving this over to CI, ran into a couple of issues accessing past the calloc'ed buffers.
- test_dataloader.c - added some const values to quiet warnings
- test_dataloader.c - added unistd.h for windows
- unistd.h - small cleanup - remove warnings for windows

CI passes.

@karpathy @ngc92  - can you please check the changes in test_shuffled and test_multiprocess_shuffled. We were accessing one token past the end of the buffer. Thank you.

Revised with comments below.

### CODE REVIEW

File: dev/unistd.h [hunk 1]
```diff
@@ -24,7 +24,10 @@ static inline int clock_gettime(int ignore_variable, struct timespec* tv)
 #define TURN_OFF_FP_FAST __pragma(float_control( precise, on, push )) // Save current setting and turn on /fp:precise
 #define TURN_ON_FP_FAST  __pragma(float_control(pop)) // Restore file's default settings
 
-#include <direct.h> /* for _mkdir and _stat */
+
```
Comment (ngc92) [inline dev/unistd.h:27]: any reason these includes are not ad the top?

Comment (rosslwheeler) [reply dev/unistd.h:27]: Yes, they can be moved up.

File: dev/test/test_dataloader.c [hunk 2]
```diff
@@ -252,7 +258,13 @@ void test_multiprocess_shuffled(void) {
         checkEquals(num_seen_inputs + start + tokens_fit, num_tokens - tokens_fit, 0);
         // verify the target counts. same thing but offset by 1
         checkEquals(num_seen_targets + start + 1, tokens_fit, num_epochs);
-        checkEquals(num_seen_targets + start + 1 + tokens_fit, num_tokens - tokens_fit, 0);
+        if (s == (num_shards - 1)) {
+            // last shard should have one less target token
+            checkEquals(num_seen_targets + start + 1 + tokens_fit, num_tokens - tokens_fit - 1, 0);
```
Comment (karpathy) [inline dev/test/test_dataloader.c:263]: I usually prefer trinary statements, e.g. `range_len = s == (num_shards -1) ? blah : bleh`, looks a bit cleaner.

Comment (rosslwheeler) [reply dev/test/test_dataloader.c:263]: Got it - will make the changes.

File: dev/unistd.h (hunk 1 above)
Comment (rosslwheeler) [reply dev/unistd.h]: Done.

File: dev/test/test_dataloader.c (hunk 2 above)
Comment (rosslwheeler) [reply dev/test/test_dataloader.c]: Done.

**Pull request merged**